### PR TITLE
[BUG] change default RNG type back to philox

### PR DESCRIPTION
**Pull request by MatthiasKohl**
As explained in code comments, it seems like we are seeing failures in RAFT DivideTests as well as cugraph RMAT tests due to usage of PC generator versus Philox.
I'm not sure whether we confirmed that the DivideTests issues are due to the generator, but at least in the case of cugraph RMAT tests, @kaatish confirmed that this was due to the generator type.

@vinaydes will be investigating these issues, but for the time being, it may be more cautious to switch back the default type to philox, and let users use PC on a case-by-case basis if performance improvements are necessary.

### CODE REVIEW

File: cpp/include/raft/random/rng_state.hpp [hunk 1]
```diff
@@ -44,10 +48,11 @@ struct RngState {
   uint64_t seed{0};
   uint64_t base_subsequence{0};
   /**
-   * The generator type. PCGenerator has been extensively tested and is faster
-   * than Philox, thus we use it as the default.
+   * The generator type. For now, use Philox as default since PCGenerator
```
[the remark this answers is not in the document]
Comment (MatthiasKohl) [reply cpp/include/raft/random/rng_state.hpp:51]: makes sense, will do

[the remark this answers is not in the document]
Comment (MatthiasKohl) [reply cpp/include/raft/random/rng_state.hpp]: I hope it's clearer now, I added a FIXME

**Review by cjnolet — DISMISSED** [review]

**Comment by MatthiasKohl** [comment]
@gpucibot merge

**Comment by cjnolet** [comment]
@MatthiasKohl if i understand the plan correctly, we're going to revert the change to the default back in this PR but keep the rng state doc that mentions the difference in perf w/ the philox vs pc generators?

**Comment by MatthiasKohl** [comment]
@cjnolet Yes, if @vinaydes is able to fix the issue in cugraph, I'll revert the change of default generator type, update the docs here w.r.t. the default and keep the part about performance.
If we do find some issue with PC which would take longer to solve, we can merge the PR as-is and later change back the default once the issue is fixed.

**Comment by cjnolet** [comment]
@MatthiasKohl this PR has gotten quite old. Should we just go ahead and close it or do you still see a reason we should merge this?

**Comment by MatthiasKohl** [comment]
@cjnolet yes I agree, let's close this, there doesn't seem to be any good reason anymore to have philox as a default RNG type

**Pull request closed without merging**